pkg/controller/rollout/rollout_bluegreen.go

```diff
@@ -305,7 +309,27 @@ func (m *blueGreenReleaseManager) doCanaryFinalising(c *RolloutContext) (bool, e
 	switch blueGreenStatus.FinalisingStep {
 	// set workload.pause=false; set workload.partition=0
 	case v1beta1.FinalisingStepResumeWorkload:
-		retry, err = finalizingBatchRelease(m.Client, c)
+		// If the rollout is being deleted and the feature gate is enabled,
+		// we skip this step entirely to keep the deployment paused.
+		if c.FinalizeReason == v1beta1.FinaliseReasonDelete && utilfeature.DefaultMutableFeatureGate.Enabled(feature.KeepDeploymentPausedOnDeletionGate) {
+			klog.Infof("Rollout(%s/%s) is being deleted, KeepDeploymentPausedOnDeletion is enabled. Skipping resume workload step.", c.Rollout.Namespace, c.Rollout.Name)
+			// Do nothing, effectively skipping this step and considering it "done".
+		} else {
```

Comment:
we cannot simply change deployment here, instead, we should change the Finalize method for the following packages:
1. pkg/controller/batchrelease/control/partitionstyle
2. pkg/controller/batchrelease/control/canarystyle
3. pkg/controller/batchrelease/control/bluegreenstyle